test/compiler/codegen.jl

```diff
@@ -500,3 +500,14 @@ end
 let f(@nospecialize(x)) = x===Base.ImmutableDict(Int128=>:big)
     @test !f(Dict(Int=>Int))
 end
+
+# issue #37974
+primitive type UInt24 24 end
+let a = Core.Intrinsics.trunc_int(UInt24, 3),
+    f(t) = t[2]
+    @test f((a, true)) === true
+    @test f((a, false)) === false
+    @test sizeof(Tuple{UInt24,Bool}) == 8
+    @test sizeof(UInt24) == 3
+    @test sizeof(Union{UInt8,UInt24}) == 3
+end
```

Comment:
```suggestion
    @test sizeof(Base.RefValue{Union{UInt8,UInt24}})) == 5 # or is this still 4?
end
```

Reply:
It's 8, since the size of the struct overall has to be aligned. Very annoying situation.